pkg/cache/cache.go

```diff
@@ -986,3 +989,30 @@ func (c *Cache) MatchingClusterQueues(nsLabels map[string]string) sets.Set[kueue
 func queueKey(q *kueue.LocalQueue) queue.LocalQueueReference {
 	return queue.NewLocalQueueReference(q.Namespace, kueue.LocalQueueName(q.Name))
 }
+
+// AddOrUpdateDynamicResourceAllocationConfig ingests the singleton CRD and updates
+// the in-memory DeviceClass → logical resource map.
+// It is idempotent and thread-safe.
+func (c *Cache) AddOrUpdateDynamicResourceAllocationConfig(log logr.Logger, draCfg *kueuealpha.DynamicResourceAllocationConfig) {
+	c.Lock()
+	defer c.Unlock()
+
+	// Rebuild from scratch to avoid stale entries.
+	newMap := make(map[corev1.ResourceName]corev1.ResourceName)
```

Comment:
```suggestion
	deviceClassNameToResName := make(map[corev1.ResourceName]corev1.ResourceName)
```

Semantic is important for variable name.